docs/SPECIFICATION.md

```diff
@@ -277,19 +280,27 @@ message CastId {
   uint64 fid = 1;                           // Fid of the cast's author
   bytes hash = 2;                           // Message.hash value of the cast
 }
+
+message Embed {
+  oneof embed {
+    string url = 1;
+    CastId cast_id = 2;
+  }
+}
 ```
 
 A CastAddBody in a message `m` is valid only if it passes these validations:
 
 1. `m.signature_scheme` must be `SIGNATURE_SCHEME_ED25519`.
 2. `m.data.type` must be `MESSAGE_TYPE_CAST_ADD`.
 3. `m.data.body.type` must be `CastAddBody`.
-4. `m.data.body.embeds` can contain up to 2 valid utf8 strings are between 1 and 256 bytes inclusive.
+4. `m.data.body.embeds_deprecated` can contain up to 2 valid utf8 strings are between 1 and 256 bytes inclusive.
 5. `m.data.body.mentions` must contain between 0 and 10 256-bit integer values.
-6. `m.data.body.parent`, if present, must be a valid CastId
+6. `m.data.body.parent`, if present, must be a valid CastId or a utf8 string that is between 1 and 256 bytes inclusive.
 7. `m.data.body.text` must contain <= 320 bytes and be a valid utf8 string.
 8. `m.data.body.mentions_positions` must have unique integers between 0 and length of `text` inclusive.
 9. `m.data.body.mentions_positions` integers must be in ascending order and must have as many elements as `mentions`.
+10. `m.data.body.embeds` can contain up to 2 CastIds or valid utf8 strings that are between 1 and 256 bytes inclusive.
```

Comment:
Nit: re-write this slightly to make it clear it can be a mix of cast ids and urls. `can contain up to 2 embeds, each of which is a CastId or valid utf8 string that is between 1 and 256 bytes inclusive in length.`